airflow/models.py

```diff
@@ -895,6 +905,8 @@ def are_dependencies_met(
         if not main_session:
             session.commit()
             session.close()
+        if verbose:
+            logging.warning("Trigger rule `{}` not satisfied".format(tr))
```

Comment:
Right, it'd be nice to show the details but would cost. This query above is optimize for a single scan and return the minimum amount of info to be able to do the job...

I'm planning to get in this area of the code and change things a bit eventually to power a UI page detailing the dependencies and explaining which ones are met and which ones aren't. This would answer clearly the most common question "Why isn't my task running?"